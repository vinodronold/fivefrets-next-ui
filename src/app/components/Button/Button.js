import React from 'react'
import { node } from 'prop-types'
import { color } from '../../constants/theme'

const _propTypes = {
  children: node.isRequired
}

const Button = ({ disabled = false, children, ...props }) => (
  <button disabled={disabled} className={`${!disabled ? 'primary' : 'disabled'}`} {...props}>
    {children}
    <style jsx>
      {`
        button {
          font-size: 0.938rem;
          cursor: pointer;
          padding: 0.5rem 1rem;
          letter-spacing: 0.2rem;
          line-height: 0.938rem;
          height: 2.5rem;
          background: transparent;
          border: none;
          transition: all 0.5s;
        }
        button.primary {
          color: ${color.primary()};
        }
        button:focus {
          outline: none;
        }
        button.primary:hover {
          color: ${color.secondary()};
        }
        button:disabled {
          cursor: not-allowed;
        }
      `}
    </style>
  </button>
)

Button.propTypes = _propTypes

export default Button
