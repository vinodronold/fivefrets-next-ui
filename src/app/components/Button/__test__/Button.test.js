import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

describe('components/Button.js', () => {
  it('should match primary snapshot', () => {
    const _primary_button = renderer.create(<Button>Button</Button>)
    expect(_primary_button).toMatchSnapshot()
  })
  it('should match disabled snapshot', () => {
    const _disabled_button = renderer.create(<Button disabled>Button Disabled</Button>)
    expect(_disabled_button).toMatchSnapshot()
  })
})
