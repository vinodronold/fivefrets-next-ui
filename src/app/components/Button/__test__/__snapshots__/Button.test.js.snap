// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button.js should match disabled snapshot 1`] = `
<button
  className="disabled"
  data-jsx={1805565103}
  disabled={true}
>
  Button Disabled
</button>
`;

exports[`components/Button.js should match primary snapshot 1`] = `
<button
  className="primary"
  data-jsx={1805565103}
  disabled={false}
>
  Button
</button>
`;
